### PR TITLE
(Part 2)Fix diffs in compute forwarding rule: Labels can only be added or modified by the setLabels method

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

Cherry-picked from #2711

Based on #2720, check the last commit

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingrule/_http.log [hunk 1]
```diff
@@ -660,14 +660,10 @@ X-Xss-Protection: 0
   "id": "000000000000000000000",
   "kind": "compute#forwardingRule",
   "labelFingerprint": "abcdef0123A=",
-  "labels": {
-    "cnrm-test": "true",
-    "label-one": "value-one",
-    "managed-by-cnrm": "true"
-  },
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingrule/_http.log:667]: qq: is there follow up PR to fix this?

File: pkg/controller/direct/compute/forwardingrule_controller.go [hunk 2]
```diff
@@ -333,11 +334,14 @@ func (a *forwardingRuleAdapter) Create(ctx context.Context, createOp *directbase
...
 		return fmt.Errorf("creating ComputeForwardingRule %s: %w", a.fullyQualifiedName(), err)
 	}
-	err = op.Wait(ctx)
-	if err != nil {
-		return fmt.Errorf("waiting ComputeForwardingRule %s create failed: %w", a.fullyQualifiedName(), err)
+	if !op.Done() {
+		err = op.Wait(ctx)
+		if err != nil {
+			return fmt.Errorf("waiting ComputeForwardingRule %s create failed: %w", a.fullyQualifiedName(), err)
+		}
```
Comment (yuwenma) [inline pkg/controller/direct/compute/forwardingrule_controller.go:341]: Mind explaining what the GCP level difference is for this change?

File: pkg/controller/direct/compute/forwardingrule_controller.go [hunk 3]
```diff
@@ -358,6 +362,55 @@ func (a *forwardingRuleAdapter) Create(ctx context.Context, createOp *directbase
 		return fmt.Errorf("getting ComputeForwardingRule %q: %w", a.fullyQualifiedName(), err)
 	}
```
Comment (yuwenma) [inline pkg/controller/direct/compute/forwardingrule_controller.go:363]: Per our discussion, I think we want to update the forwarding rule status here, right?

File: pkg/controller/direct/compute/forwardingrule_controller.go (hunk 2 above)
Comment (justinsb) [reply pkg/controller/direct/compute/forwardingrule_controller.go:341]: I think I know this one from chatting with @gemmahou ; the issue is that we don't want to get the operation if it is already done - we don't want an extra operation get in the http log.

File: pkg/controller/direct/compute/forwardingrule_controller.go (hunk 3 above)
Comment (justinsb) [reply pkg/controller/direct/compute/forwardingrule_controller.go:363]: I think that's blocked on my getting my status changes merged though - I'll work on that!

File: pkg/controller/direct/compute/forwardingrule_controller.go [hunk 4]
```diff
@@ -391,49 +444,33 @@ func (a *forwardingRuleAdapter) Update(ctx context.Context, updateOp *directbase
...
-		}
-		op, err = a.globalForwardingRulesClient.SetLabels(ctx, setLabelsReq)
-	} else {
-		setLabelsReq := &computepb.SetLabelsForwardingRuleRequest{
-			Resource:                       a.id.forwardingRule,
-			RegionSetLabelsRequestResource: &computepb.RegionSetLabelsRequest{LabelFingerprint: a.actual.LabelFingerprint, Labels: forwardingRule.Labels},
-			Project:                        a.id.project,
-			Region:                         a.id.location,
+	if !reflect.DeepEqual(forwardingRule.Labels, a.actual.Labels) {
+		if a.id.location == "global" {
```
Comment (justinsb) [inline pkg/controller/direct/compute/forwardingrule_controller.go:448]: Nit: looks like we could refactor out a `a.setLabels` helper to get rid of the duplicated setLabels code ?

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingrule/_http.log (hunk 1 above)
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingrule/_http.log:667]: We do call setLabels below...

**Comment by justinsb** [comment]
/lgtm

I'll leave approval to @yuwenma

File: pkg/controller/direct/compute/forwardingrule_controller.go (hunk 2 above)
Comment (gemmahou) [reply pkg/controller/direct/compute/forwardingrule_controller.go:341]: Right!

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingrule/_http.log (hunk 1 above)
Comment (gemmahou) [reply pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingrule/_http.log:667]: This is expected: 
We first create the forwarding rule without labels, and then call setLabels to add labels. So I removed labels field when the resource is created in mock server, and let setLabels to add it back-that matches the realGCP.

**Comment by gemmahou** [comment]
Rebased and moved `get` and `setLabels` to their helper function

**Pull request merged**